pymagicc/io.py

```diff
@@ -1045,13 +1045,28 @@ def find_region(x):
 
 
 class _InputWriter(object):
-    # need this for the _get_initial_nml_and_data_block routine as SCEN7
-    # files have a special, contradictory set of region flags
-    # would be nice to be able to remove in future
+    """Base class for writing MAGICC input files.
+
+    Attributes
+    ----------
+    _magicc_version : int
+        The MAGICC version for which we want to write files. MAGICC7 and MAGICC6
+        namelists are incompatible hence we need to know which one we're writing for.
+
+    _scen_7 : bool
+        Whether this writer writes SCEN7 files or not. We need this as SCEN7 files
+        have a unique definition of what the THISFILE_REGIONMODE flag should be set to
+        which is contradictory to all other files.
+    """
+
+    _magicc_version = 7
```

Comment:
Why are here both a magicc and a scen file flag?

Reply:
do the comments not explain it?

Reply:
yes it can :'(